python/main.py

```diff
@@ -4,39 +4,104 @@
 from fastapi import FastAPI, Form, HTTPException
 from fastapi.responses import FileResponse
 from fastapi.middleware.cors import CORSMiddleware
+import db
+import hashlib
+from os.path import join, dirname, realpath
 
 app = FastAPI()
 logger = logging.getLogger("uvicorn")
-logger.level = logging.INFO
+logger.level = logging.DEBUG
 images = pathlib.Path(__file__).parent.resolve() / "image"
-origins = [ os.environ.get('FRONT_URL', 'http://localhost:3000') ]
+origins = [os.environ.get('FRONT_URL', 'http://localhost:3000')]
 app.add_middleware(
     CORSMiddleware,
     allow_origins=origins,
     allow_credentials=False,
-    allow_methods=["GET","POST","PUT","DELETE"],
+    allow_methods=["GET", "POST", "PUT", "DELETE"],
     allow_headers=["*"],
 )
 
+
 @app.get("/")
 def root():
     return {"message": "Hello, world!"}
 
+
+@app.get("/items")
+async def read_items():
+    items = db.get_items()
+    all_items = []
+    for item in items:
+        all_items.append(
+            {"id": item[0], "name": item[1], "category": item[2], "image": item[3]})
+    return all_items
+
+
+@app.get("/items/{item_id}")
+async def read_item(item_id: int):
+    item = db.get_item(item_id)
+    if not item:
+        raise HTTPException(status_code=404, detail="Item not found")
+    return {"id": item[0], "name": item[1], "category": item[2], "image": item[3]}
+
+
+@app.get("/search")
+async def read_items(keyword: str):
+    all_items = {"items": []}
+    items = db.search_items(keyword)
+    for item in items:
+        all_items["items"].append(
+            {"id": item[0], "name": item[1], "category": item[2], "image": item[3]})
+    logger.info(f"{all_items}")
+    return all_items
+
+
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+def add_item(name: str = Form(...), category: str = Form(...), image: str = Form(...)):
```

Comment:
Instead of sending the file name, you could send the image file itself as bytes 🙂 like "image: bytes = File(...)"